File: Python/Problem_1578.py
from typing import List

# Problem - 1578: https://leetcode.com/problems/minimum-time-to-make-rope-colorful/
# Python3 Solution!
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        first_baloon, result = 0 , 0

        for second_baloon in range(1, len(colors)):
            if colors[first_baloon] == colors[second_baloon]:
                if neededTime[first_baloon] < neededTime[second_baloon]:
                    result += neededTime[first_baloon]
                    first_baloon = second_baloon
                else:
                    result += neededTime[second_baloon]
            else:
                first_baloon = second_baloon

        return result


if __name__=="__main__":
    sol = Solution()

    print(sol.minCost("abaac", [1,2,3,4,5]))
    print(sol.minCost("abc", [1,2,3]))
    print(sol.minCost("aabaa", [1,2,3,4,1]))

File: Java/Problem_1578/Problem_1578.java
package Java.Problem_1578;
/* 
* Problem - 1578: https://leetcode.com/problems/minimum-time-to-make-rope-colorful/
* Java Solution!
*/
class Solution {
    public int minCost(String colors, int[] neededTime) {
        int first_baloon = 0, result = 0;

        for (int second_baloon = 1; second_baloon < colors.length(); second_baloon++){
            if (colors.charAt(first_baloon) == colors.charAt(second_baloon)) {
                if (neededTime[first_baloon] < neededTime[second_baloon]){
                    result += neededTime[first_baloon];
                    first_baloon = second_baloon;
                }
                else {
                    result += neededTime[second_baloon];
                }
            }
            else {
                first_baloon = second_baloon;
            }
        }

        return result;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println(sol.minCost("abaac", new int[] {1,2,3,4,5}));
        System.out.println(sol.minCost("abc", new int[] {1,2,3}));
        System.out.println(sol.minCost("aabaa", new int[] {1,2,3,4,1}));
    }
}